Sources/ClientRuntime/Telemetry/Metrics/Meter.swift

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// A Meter is the entry point to create the instruments.
+///
+/// The following instrument creation entry points are defined:
+///
+/// - Gauge
+/// - UpDownCounter
+/// - Asynchronous UpDownCounter
+/// - MonotonicCounter
+/// - Asynchronous MonotonicCounter
+/// - Histogram
+public protocol Meter {
+    /// Creates a Gauge, used to measure the current instantaneous value of something.
+    ///
+    /// Examples include: the current memory used by a process
+    ///
+    /// - Parameter name: the instrument name
+    /// - Parameter callback: callback invoked when a Gauge value is read
+    /// - Parameter units: the unit of measure
+    /// - Parameter description: a description of the metric
+    /// - Returns: handle to stop recording Gauge metrics
+    func createGauge(
+        name: String,
+        callback: @escaping (any DoubleAsyncMeasurement) -> Void,
+        units: String?,
+        description: String?
+    ) -> AsyncMeasurementHandle
+    /// Creates an UpDownCounter, used to measure a value that goes up or down.
```

Comment:
nit: Let's add 1-line vertical spacings between function signatures for readability.

Reply:
The newlines were added between function signatures.